openfga/v1/authzmodel.proto

```diff
@@ -52,19 +52,11 @@ message TypeDefinition {
 }
 
 message Relation {
-  string name = 1 [(validate.rules).string = {
-    pattern: "^[^:#@\\s]{1,50}$"
-  }];
-
   Userset rewrite = 2 [
     (validate.rules).message.required = true,
     (google.api.field_behavior) = REQUIRED
   ];
 
-  RelationTypeInfo type_info = 3;
```

Comment:
RelationTypeInfo may expand to include other additional metadata in the future, and so I believe we should leave it. Another reason is clarity. DirectlyRelatedUserTypes define metadata specific to the TypeInfo or TypeRestrictions of the Relation, and so I believe it makes the most sense to embed them in a type that represents/encompasses that.

It is also easier to evolve an object type that embodies a certain abstraction than it is inline definitions. For example, if we add other metadata as part of the type system at some point, then we'd just append it to the Relation and we'd have to join ti with the other fields that are applicable to type info. With the current approach we simply `relation.GetTypInfo()` to get anything type info related. Otherwise we'd have to do something like `relation.GetDirectlyRelatedUserTypes()` and `relation.GetOtherTypeInfoMetadata()` etc..